app/util/rect.ts

```diff
@@ -0,0 +1,66 @@
+import { v2, Vec2 } from './vec2';
+import { ServiceHelper } from '../services/stateful-service';
+
+/**
+ * Class for a simple rectangle
+ */
+@ServiceHelper()
+export class Rect implements IRectangle {
```

Comment:
the reason I created this class is because of I want to have a simple class for operations with rectangles that has no deal with fields like `croppedWidth` and `scaledWidth`